libs/email-service/src/main/java/uk/gov/di/ipv/core/library/service/EmailService.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.di.ipv.core.library.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.config.ConfigurationVariable;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EmailService {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final int NUMBER_OF_RETRIES = 3;
+    private static final int RETRY_WAIT_MILLISECONDS = 3000;
+
+    private final ConfigService configService;
+    private final NotificationClient notificationClient;
+    private final int retryWaitInMilliseconds;
+
+    @ExcludeFromGeneratedCoverageReport
+    public EmailService(
+            ConfigService configService,
+            NotificationClient notificationClient,
+            int retryWaitInMilliseconds) {
+        this.configService = configService;
+        this.notificationClient = notificationClient;
+        this.retryWaitInMilliseconds = retryWaitInMilliseconds;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public EmailService(ConfigService configService, NotificationClient notificationClient) {
+        this(configService, notificationClient, RETRY_WAIT_MILLISECONDS);
+    }
+
+    public void sendUserTriggeredIdentityResetConfirmation(
+            String userEmailAddress, String fullName) {
+        Map<String, Object> templateParameters = new HashMap<>();
+        templateParameters.put("fullName", fullName);
+
+        LOGGER.info("Attempting to send user triggered identity reset confirmation email");
+        // This template ID can vary between production and the other environments, so it can't be
+        // hardcoded
+        final String templateId =
+                configService.getSsmParameter(
+                        ConfigurationVariable
+                                .GOV_UK_NOTIFY_TEMPLATE_ID_USER_TRIGGERED_IDENTITY_RESET_CONFIRMATION);
+        LOGGER.info("Got template ID {}", templateId);
+        SendEmail(templateId, userEmailAddress, templateParameters);
+    }
+
+    private void SendEmail(
+            String templateId, String toAddress, Map<String, Object> personalisation) {
+
+        var retries = 0;
+
+        while (true) {
+            try {
+                LOGGER.info("About to send email");
+                notificationClient.sendEmail(templateId, toAddress, personalisation, null, null);
+                LOGGER.info("Email sent");
```

Comment:
I'd probably make these debug level (or remove)?

Reply:
Good point, these were inserted when trying to find out exactly what was going wrong.